tools/release/ltsgraph/utility.h

```diff
@@ -68,7 +68,20 @@ constexpr float PI_2 = PI * 0.5f;
 inline
 float frand(float min, float max)
 {
-  return static_cast<float>(QRandomGenerator::global()->generateDouble() * (max - min) + min);
+  static thread_local std::random_device rd;
+  static thread_local std::mt19937 twister(rd());
+  std::uniform_real_distribution<> dist(min, max);
+  return dist(twister);
+}
+
+inline
+float fast_frand(float min, float max)
+{
+  // Fast pseudo rand, source: http://www.musicdsp.org/showone.php?id=273
+  static thread_local std::random_device rd;
+  static int32_t seed = rd();
+  seed *= 16807;
```

Comment:
Note that this multiplication is still not thread safe. You could claim though that it doesn't matter much if a race condition occurs and `seed` is filled with a random number ;-). Nice use of thread local static variables by the way.

Reply:
Ha good catch, I missed that. Yes, this is the only time we ever want random garbage ;) Fixed it though :)